Menus/ObjectPickerGUI.cpp

```diff
@@ -25,1296 +12,559 @@
-            // Clicks on the Objects List
-
-            else if (anEvent.GetControl() == m_pObjectsList)
-            {
-/*
-                // Somehting was just selected/highligted, so update the selection index to the new selected index
-                if(anEvent.GetMsg() == GUIListBox::Select)
-                {
-                    if (m_SelectedObjectIndex != m_pObjectsList->GetSelectedIndex())
-                        g_GUISound.SelectionChangeSound()->Play(m_pController->GetPlayer());
-                    m_SelectedObjectIndex = m_pObjectsList->GetSelectedIndex();
-                }
-                // MouseDown, picked somehting
-                else*/ if (anEvent.GetMsg() == GUIListBox::MouseDown)
-                {
-                    GUIListPanel::Item *pItem = m_pObjectsList->GetSelected();
-                    m_pObjectsList->ScrollToSelected();
-
-                    // If a module group list item, toggle its expansion and update the list
-                    if (pItem && pItem->m_ExtraIndex >= 0)
-                    {
-                        // Make appropriate sound
-                        if (!m_aExpandedModules[pItem->m_ExtraIndex])
-                            g_GUISound.ItemChangeSound()->Play(m_pController->GetPlayer());
-                        // Different, maybe?
-                        else
-                            g_GUISound.ItemChangeSound()->Play(m_pController->GetPlayer());
-                        // Toggle the expansion of the module group item's items below
-                        m_aExpandedModules[pItem->m_ExtraIndex] = !m_aExpandedModules[pItem->m_ExtraIndex];
-                        // Re-populate the item list with the new module expansion configuation
-                        UpdateObjectsList(false);
-                    }
-                    // Only add if there's an entity attached to the list item
-                    else if (pItem && pItem->m_pEntity)
-                    {
-                        m_SelectedObjectIndex = m_pObjectsList->GetSelectedIndex();
-                        // User has made a pick, so close the Picker
-                        if (m_pPickedObject = dynamic_cast<const SceneObject *>(pItem->m_pEntity))
-                        {
-                            g_GUISound.ObjectPickedSound()->Play(m_pController->GetPlayer());
-                            SetEnabled(false);
-                        }
-                    }
-                    // Undo the click deselection if nothing was selected
-//                    else
-//                        m_pObjectsList->SetSelectedIndex(m_SelectedObjectIndex);
-
-                    m_PickerFocus = OBJECTS;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ObjectPickerGUI::SelectGroupByIndex(int groupIndex, bool updateObjectsList) {
+		m_SelectedGroupIndex = (groupIndex < 0) ? m_ShownGroupIndex : groupIndex;
```

Comment:
I'm a little confused by the `m_SelectedGroupIndex = (groupIndex < 0) ? m_ShownGroupIndex : groupIndex;` safety check thing and wanna be sure I'm understanding its use. Is it there to allow you to put in -1 as the groupIndex and just select the shownGroupIndex? I guess this is used in mouse handling, so if you click on nothing or move your mouse off a group entry it'll just default to selecting the shown group index?

Assuming this is the case, the method comment should explain this a bit. Probably have a note on the `groupIndex` param saying if it's <0 this will select the currently shown group or whatever.